### PR TITLE
Strict compiler settings for discovery

### PR DESCRIPTION
**Pull request by chbatey**

### CODE REVIEW

File: project/AkkaBuild.scala [hunk 1]
```diff
@@ -31,6 +31,10 @@ object AkkaBuild {
       version in ThisBuild := "2.5-SNAPSHOT"
     )
 
+  lazy val strictSettings = Seq(
```
Comment (raboof) [inline project/AkkaBuild.scala]: Hmm, wouldn't this belong in `project/AkkaDisciplinePlugin`?

**Review by raboof — APPROVED** [review]

File: project/AkkaBuild.scala (hunk 1 above)
Comment (helena) [reply project/AkkaBuild.scala]: Yes, that is where it belongs.

Comment (chbatey) [reply project/AkkaBuild.scala]: I hadn't seen that plugin, I'll just change it to have modules where it doesn't remove the fatal wanrings

**Review by raboof — APPROVED** [review]
Seems neat. At some point we can change from default-loose to default-strict ;)

**Review by helena — APPROVED** [review]
LGTM!

**Pull request merged**